File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        m = {i:nums.count(i) for i in set(nums)}
        m = sorted(m.items(), reverse=True, key=lambda kv: kv[1])
        return map(lambda kv: kv[0], m[:k])

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> mp;
        vector<int> ans;
        priority_queue<pair<int,int>> pq;

        for(int i=0;i<nums.size();i++)
        {
            mp[nums[i]]++;
        }

        for(auto it=mp.begin();it!=mp.end();it++)
        {
            pq.push(make_pair(it->second, it->first));

            if(pq.size() > (int)mp.size() - k)
            {
                ans.push_back(pq.top().second);
                pq.pop();
                
            }
        }
        return ans;
    }
};